mongodump/mongodump.go

```diff
@@ -513,6 +516,7 @@ func (dump *MongoDump) dumpIterToWriter(
 			return fmt.Errorf("error writing to file: %v", err)
 		}
 		progressCount.Inc(1)
+		dump.count += 1
```

Comment:
You're right that this could be better. The two interfaces are left over from a design plan that ended up changing. At this point, I'd recommend merging both interfaces into one (just keep the name "Progressor"). This might require some very minor updates for the other tools as well.